pkg/collector/corechecks/cluster/ksm/customresources/apiservice.go

```diff
@@ -0,0 +1,147 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+// +build kubeapiserver
+
+package customresources
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	v1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	apiregistrationclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
+	"k8s.io/kube-state-metrics/v2/pkg/customresource"
+	"k8s.io/kube-state-metrics/v2/pkg/metric"
+	generator "k8s.io/kube-state-metrics/v2/pkg/metric_generator"
+)
+
+var (
+	descApiServiceAnnotationsName     = "kube_apiservice_annotations"
+	descApiServiceAnnotationsHelp     = "Kubernetes annotations converted to Prometheus labels."
+	descApiServiceLabelsName          = "kube_apiservice_labels"
+	descApiServiceLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descApiServiceLabelsDefaultLabels = []string{"apiservice"}
+)
+
+// NewAPIServiceFactory returns a new APIService metric family generator factory.
+func NewAPIServiceFactory(client *apiserver.APIClient) customresource.RegistryFactory {
+	return &apiserviceFactory{
+		client: client.APISClient,
+	}
+}
+
+type apiserviceFactory struct {
+	client interface{}
+}
+
+// CreateClient is not implemented
```

Comment:
```suggestion
```

Reply:
Makes sense! To avoid adding "un-related" files to the PRs, I did not remove the comment from the other custom resources, but same logic would apply